databuilder/databuilder/task/search/document_mappings.py

```diff
@@ -57,31 +57,55 @@ class Analyzer:
                                                   Filter.english_stop,
                                                   Filter.english_stemmer])
 
+
+class Subfield:
+    # combinations of field types and analyzers for additional index time analysis
+
+    keyword = Keyword()
+
+    alphanumeric = Text(analyzer=Analyzer.alphanum_analyzer,
+                        term_vector=POSITIONS_OFFSETS)
+
+    general = Text(analyzer=Analyzer.general_analyzer,
+                   term_vector=POSITIONS_OFFSETS)
+
+    general_multi = Text(multi=True,
+                         analyzer=Analyzer.general_analyzer,
+                         term_vector=POSITIONS_OFFSETS)
+
+    alphanumeric_multi = Text(multi=True,
+                              analyzer=Analyzer.alphanum_analyzer,
+                              term_vector=POSITIONS_OFFSETS)
+
 # Resource Mappings
 
 
 class SearchableResource(Document):
     # For better understanding of field type rationale read "Mapping unstructured content"
     # https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html#wildcard-field-type
     key = Text(required=True,
-               fields={"keyword": Keyword()},
+               fields={"keyword": Subfield.keyword},
```

Comment:
this is the same instance every time - is that okay?

Reply:
it doesn't matter because what elasticsearch-dsl does is it turns this into json and makes the request

Reply:
if you look at the analyzers and stuff too in the library `analyzer = Analyzer._type_shortcut`